### PR TITLE
New devtool to update testdata

### PR DESCRIPTION
**Pull request by theresantonie**
### Short description
 
Add a new `.sh` devtool that updates the testdata file based on the current database to quickly update the data while writing tests.

### Proposed changes

- Adding the new `.sh` tool `tools/update_test_data.sh`
- Adding documentation in `docs/src/tools.rst`

### Side effects

- None

### Resolved issues

Fixes: #2793

__________________________________________________

[Pull Request Review Guidelines](https://digitalfabrik.github.io/integreat-cms/pull-request-review-guide.html)

### CODE REVIEW

File: integreat_cms/release_notes/current/unreleased/2793.yml
Comment (deen13) [inline integreat_cms/release_notes/current/unreleased/2793.yml:2]: I believe we should skip the release note for this pull request. The developers will likely see the pull request directly, and including a note might confuse the users. 😄

Comment (theresantonie) [reply integreat_cms/release_notes/current/unreleased/2793.yml:2]: Ah, you are right. 😄 Thanks!

File: tools/update_test_data.sh
Comment (david-venhoff) [inline tools/update_test_data.sh:10]: Did not test this yet, but I think the output should replace the existing test data file at `integreat_cms/cms/fixture/test_data.json`. Right now it only creates a new file in the root directory.

Comment (theresantonie) [reply tools/update_test_data.sh]: Thanks!

**Review by david-venhoff — COMMENTED** [review]
One problem with this solution is that this command will not format the new test data file, so I think that we should also automatically run a command that formats the test data. ~~@MizukiTemma @JoeyStk @charludo do you have an idea maybe what tool we could use for that?~~

File: tools/update_test_data.sh
Comment (david-venhoff) [inline tools/update_test_data.sh:10]:
```suggestion
deescalate_privileges integreat-cms-cli dumpdata --natural-primary --natural-foreign --exclude auth --exclude contenttypes --o /integreat_cms/cms/fixtures/test_data.json
# Format the new function
source "$(dirname "${BASH_SOURCE[0]}")/prettier.sh"
```
By running prettier we can make sure that the generated json file is consistently formatted

Comment (charludo) [reply tools/update_test_data.sh:10]: +1 for prettier, but it's probably even nicer to just call `prettier` directly on the one changed file?

File: docs/src/tools.rst
Comment (theresantonie) [reply docs/src/tools.rst]: Thank you for the suggestion. 
I have added a corresponding disclaimer to the documentation. :)

File: integreat_cms/release_notes/current/unreleased/2793.yml
Comment (charludo) [reply integreat_cms/release_notes/current/unreleased/2793.yml:2]: THis has reappeared ^^

**Review by charludo — APPROVED** [review]
Thank you, this looks good (except for the one convo that resurfaced :D)

File: integreat_cms/release_notes/current/unreleased/2793.yml
Comment (theresantonie) [reply integreat_cms/release_notes/current/unreleased/2793.yml:2]: Thanks :D

File: docs/src/tools.rst
Comment (MizukiTemma) [inline docs/src/tools.rst:70]:
```suggestion
Update testdata file based on the *current* database with :github-source:`tools/update_test_data.sh`  
**Disclaimer**: make sure to first run ``prune_database.sh`` and make *only* the desired changes before running this tool::  

    ./tools/update_test_data.sh

.. _translations:


```

**Review by MizukiTemma — CHANGES_REQUESTED** [review]
Looks good 😸 
Only one suggestion to pass the checks.

File: docs/src/tools.rst
Comment (theresantonie) [reply docs/src/tools.rst]: Thank you for the suggestion, but apparently the lineblock is necessary to have a line break before the disclaimer. 
The linter only required a newline **after** the lineblock.
I have fixed this, and the checks are passing now.

**Comment by MizukiTemma** [comment]
@theresantonie Nice 😸 Thank you for the update 👍 

Could you squash the commits? Then it's ready to go.

**Comment by theresantonie** [comment]
@MizukiTemma 

Perfect, I will "Squash and Merge" as soon as the required approvals are given, thank you for your input!

**Review by MizukiTemma — APPROVED** [review]
🎉

**Pull request merged**
